Updated content: `button` component (#910)
hashicorp/design-system: website/docs/components/button/partials/guidelines/guidelines.md

## Usage

### When to use

- To trigger an action or event, like a form submission.

### When not to use

- When navigating to another destination, use an [Inline](/components/link/inline/overview) or [Standalone](/components/link/standalone/overview) link.
- To display multiple actions under a single button, use [Dropdown](/components/dropdown/overview).

### Sizes

<Hds::Button @size="small" @text="Small" />

<Hds::Button @size="medium" @text="Medium" />

<Hds::Button @size="large" @text="Large" />

<Hds::Button @size="large" @isFullWidth={{true}} @text="Full-width button" />

!!! Insight

**Migrating Tip** 

Medium buttons are the same size as default Structure buttons. Small buttons are the same size as compact Structure buttons.

!!!

Medium is the preferred size, but use a button size that best fits the UI (i.e., don't use a `large` button in a table).

Buttons are able to be manually stretched to accommodate a need for a full-width button. Full-width buttons should fill 100% of the parent container.

> !!! Info

Author: ```suggestion
```

> !!!

Author: ```suggestion
```

## Usage

### When to use

- To trigger an action or event, like a form submission.

### When not to use

- When navigating to another destination, use an [Inline](/components/link/inline/overview) or [Standalone](/components/link/standalone/overview) link.
- To display multiple actions under a single button, use [Dropdown](/components/dropdown/overview).

### Sizes

<Hds::Button @size="small" @text="Small" />

<Hds::Button @size="medium" @text="Medium" />

<Hds::Button @size="large" @text="Large" />

<Hds::Button @size="large" @isFullWidth={{true}} @text="Full-width button" />

!!! Insight

**Migrating Tip** 

Medium buttons are the same size as default Structure buttons. Small buttons are the same size as compact Structure buttons.

!!!

Medium is the preferred size, but use a button size that best fits the UI (i.e., don't use a `large` button in a table).

Buttons are able to be manually stretched to accommodate a need for a full-width button. Full-width buttons should fill 100% of the parent container.

!!! Info

> **Figma Tip:** Select the `button` layer and manually adjust the width to the width needed, then select the child layer (i.e., `\_main/[size]`) and change _Resizing_ for width to 'Fill Container'

Author: ```suggestion
```

> !!!

Reviewer: It's ok to remove these if you see them, as our Figma Tips page will be staying directly in Figma.

<Hds::Button @size="large" @text="Large" />

<Hds::Button @size="large" @isFullWidth={{true}} @text="Full-width button" />

!!! Insight

**Migrating Tip** 

> Medium buttons are the same size as default Structure buttons. Small buttons are the same size as Compact Structure buttons.

Reviewer 2: Q: Should "Compact" be capitalized here?

"Small buttons are the same size as **Compact** Structure buttons."

Author: That sounds good to me- can you make it a suggestion so I can merge it from here? 👍

- for actions on a page that require even less visual weight or priority than a secondary button (our default for less important actions)
- for controlling form fields (such as adding an additional form field)

> ### State

Reviewer: The images in the State section are a bit pixelated. I recommend working with @cveigt to get higher quality assets exported.

Author: Great idea! I created a jira issue to track. https://hashicorp.atlassian.net/browse/HDS-1316

- Incomplete flows (ie. user has not yet completed a required action needed to move forward).
- Permissions restriction.

### Icon position

  <Hds::ButtonSet>
    <Hds::Button @color="secondary" @text="No icon" />
    <Hds::Button @color="secondary" @text="Leading Icon" @icon="plus" @iconPosition="leading" />
    <Hds::Button @color="secondary" @text="Trailing Icon" @icon="arrow-right" @iconPosition="trailing" />
    <Hds::Button @color="secondary" @text="No text" @icon="plus" @isIconOnly={{true}} />
  </Hds::ButtonSet>

Buttons are provided with flexible icon use; allowing for leading, trailing, or icon only buttons. Use icons intentionally and only when they provide the user with extra value. Do not create buttons with both leading and trailing icons. Tertiary buttons are required to have either a leading or trailing icon layout to be accessible.

> _Banner (highlight):_ Buttons used for a dropdown (with the Caret Icon in the Trailing Icon Spot) can be found in [Dropdown](/components/dropdown/overview).

Author: ```suggestion
!!! Insight

**Looking for Dropdowns?**

Buttons used for a dropdown (with the Caret Icon in the Trailing Icon Spot) can be found in [Dropdown](/components/dropdown/overview).

!!!
```
